src/back/spv/writer.rs

```diff
@@ -240,24 +242,35 @@ impl Writer {
                     })) {
                     Entry::Occupied(e) => *e.get(),
                     _ => {
-                        let pointer_id = self.generate_id();
-                        let instruction =
-                            super::instructions::instruction_type_pointer(pointer_id, class, ty_id);
-                        instruction.to_words(&mut self.logical_layout.declarations);
+                        let id = self.create_pointer(ty_id, class);
                         self.lookup_type.insert(
                             LookupType::Local(LocalType::Pointer {
                                 base: handle,
                                 class,
                             }),
-                            pointer_id,
+                            id,
                         );
-                        pointer_id
+                        id
                     }
                 }
             }
         }
     }
 
+    fn create_pointer(&mut self, ty_id: Word, class: spirv::StorageClass) -> Word {
+        let id = self.generate_id();
+        let instruction = super::instructions::instruction_type_pointer(id, class, ty_id);
+        instruction.to_words(&mut self.logical_layout.declarations);
+        id
+    }
+
+    fn create_constant(&mut self, type_id: Word, value: &[Word]) -> Word {
+        let id = self.generate_id();
+        let instruction = super::instructions::instruction_constant(type_id, id, value);
+        instruction.to_words(&mut self.logical_layout.declarations);
+        id
+    }
```

Comment:
Both cannot easily be stored in a lookup now without refactoring some more pieces. For now whenever the same pointer or same constant is found, it just "duplicates" it, resulting in duplicate values in SPIR-V.

I preferably don't want to block on this, but I can create an issue for it.